Packages/com.unity.inputsystem/InputSystem/Actions/InputAction.cs

```diff
@@ -608,6 +607,42 @@ public unsafe InputControl activeControl
             }
         }
 
+        /// <summary>
+        /// Type of value returned by <see cref="ReadValueAsObject"/> and expected
+        /// by <see cref="ReadValue{TValue}"/>. <see langword="null"/> while the action
+        /// is in waiting (<see cref="InputActionPhase.Waiting"/>) or canceled (<see cref="InputActionPhase.Canceled"/>)
+        /// state as this is based on the currently active control that is driving the action.
+        /// </summary>
+        /// <value>Type of object returned when reading a value.</value>
+        /// <remarks>
+        /// The type of value returned by an action is usually determined by the
+        /// <see cref="InputControl"/> that triggered the action, i.e. by the
+        /// control referenced from <see cref="activeControl"/>.
+        ///
+        /// However, if the binding that triggered is a composite, then the composite
+        /// will determine values and not the individual control that triggered (that
+        /// one just feeds values into the composite).
+        /// </remarks>
+        /// <seealso cref="InputControl.valueType"/>
+        /// <seealso cref="InputBindingComposite.valueType"/>
+        /// <seealso cref="activeControl"/>
+        public unsafe Type activeValueType
```

Comment:
I added some notes to the xml-doc. Basically we have some input actions where we permit multiple bindings of different control types. In XRI, we allow either a Vector2 or float trigger for "select", but we can't call ReadValue<float> or ReadValue<Vector2> without it potentially throwing an exception. We have to wrap it in a try/catch. It would be nicer if we could query the expected type directly to know which version to call.

This value is already exposed in the `CallbackContext` with `CallbackContext.valueType`, but this property lets us get it in a polling way instead of having to be in the `performed` callback method.

Reply:
Totally agree or potentially in the long run avoid having non-typed actions in general. Thanks for explaining.